// @elastic/eui/optimize/lib/components/skeleton/skeleton_title.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EuiSkeletonTitle = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _services = require("../../services");

var _skeleton_loading = require("./skeleton_loading");

var _skeleton_title = require("./skeleton_title.styles");

var _react2 = require("@emotion/react");

var _excluded = ["isLoading", "size", "className", "contentAriaLabel", "ariaWrapperProps", "children"];

var EuiSkeletonTitle = function EuiSkeletonTitle(_ref) {
  var _ref$isLoading = _ref.isLoading,
      isLoading = _ref$isLoading === void 0 ? true : _ref$isLoading,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 'm' : _ref$size,
      className = _ref.className,
      contentAriaLabel = _ref.contentAriaLabel,
      ariaWrapperProps = _ref.ariaWrapperProps,
      children = _ref.children,
      rest = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var euiTheme = (0, _services.useEuiTheme)();
  var styles = (0, _skeleton_title.euiSkeletonTitleStyles)(euiTheme);
  var cssStyles = [styles.euiSkeletonTitle, styles[size]];
  return (0, _react2.jsx)(_skeleton_loading.EuiSkeletonLoading, (0, _extends2.default)({
    isLoading: isLoading,
    loadingContent: (0, _react2.jsx)("span", (0, _extends2.default)({
      className: (0, _classnames.default)('euiSkeletonTitle', className),
      css: cssStyles
    }, rest)),
    loadedContent: children || '',
    contentAriaLabel: contentAriaLabel
  }, ariaWrapperProps));
};

exports.EuiSkeletonTitle = EuiSkeletonTitle;
