// @elastic/eui/test-env/components/skeleton/skeleton_loading.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EuiSkeletonLoading = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _screen_reader_live = require("../accessibility/screen_reader_live");

var _i18n = require("../i18n");

var _react2 = require("@emotion/react");

var _excluded = ["isLoading", "contentAriaLabel", "loadingContent", "loadedContent"];

var EuiSkeletonLoading = function EuiSkeletonLoading(_ref) {
  var _ref$isLoading = _ref.isLoading,
      isLoading = _ref$isLoading === void 0 ? true : _ref$isLoading,
      contentAriaLabel = _ref.contentAriaLabel,
      loadingContent = _ref.loadingContent,
      loadedContent = _ref.loadedContent,
      rest = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var loadingAriaLabel = (0, _i18n.useEuiI18n)('euiSkeletonLoading.loadingAriaText', 'Loading {contentAriaLabel}', {
    contentAriaLabel: contentAriaLabel
  });
  var loadedAriaLive = (0, _i18n.useEuiI18n)('euiSkeletonLoading.loadedAriaText', 'Loaded {contentAriaLabel}', {
    contentAriaLabel: contentAriaLabel
  });
  var loadingProps = {
    'aria-label': loadingAriaLabel,
    role: 'progressbar'
  };
  return (0, _react2.jsx)("div", (0, _extends2.default)({
    "aria-busy": isLoading,
    "data-test-subj": "euiSkeletonLoadingAriaWrapper"
  }, rest), isLoading ? /*#__PURE__*/_react.default.cloneElement(loadingContent, loadingProps) : (0, _react2.jsx)(_react.default.Fragment, null, (0, _react2.jsx)(_screen_reader_live.EuiScreenReaderLive, null, loadedAriaLive), loadedContent));
};

exports.EuiSkeletonLoading = EuiSkeletonLoading;
EuiSkeletonLoading.propTypes = {
  className: _propTypes.default.string,
  "aria-label": _propTypes.default.string,
  "data-test-subj": _propTypes.default.string,
  css: _propTypes.default.any,

  /**
       * Content to display when loading
       */
  loadingContent: _propTypes.default.element.isRequired,

  /**
       * Content to display when loaded
       */
  loadedContent: _propTypes.default.any.isRequired
};
